src/DDTrackCreatorBase.cc

```diff
@@ -421,16 +421,34 @@ void DDTrackCreatorBase::GetTrackStatesAtCalo( EVENT::Track *track,
   auto marlintrk = std::unique_ptr<MarlinTrk::IMarlinTrack>(m_trackingSystem->createTrack());
   const EVENT::TrackerHitVec& trkHits = track->getTrackerHits();
   const int nHitsTrack = trkHits.size();
+
   for (int iHit = 0; iHit < nHitsTrack; ++iHit) {
-      marlintrk->addHit(trkHits[iHit]);
+    EVENT::TrackerHit* trkHit = trkHits[iHit] ;
+    if( UTIL::BitSet32( trkHit->getType() )[ UTIL::ILDTrkHitTypeBit::COMPOSITE_SPACEPOINT ]   ){ //it is a composite spacepoint
```

Comment:
Why is this necessary, when we get all the hits from the original track? Weren't the hits of the track already added the same way?

Reply:
No, this is what you have to do to pass an lcio::TrackerHit that is a spacepoint made from two strip hits to MarlinTrk - see here:
https://github.com/iLCSoft/MarlinTrk/blob/master/src/MarlinTrkUtils.cc#L236-L260

Reply:
Then it would be better to change this in the `addHit` function itself. Avoiding all this code duplication